inlanecruising_plugin/src/inlanecruising_plugin.cpp

```diff
@@ -267,4 +267,89 @@ namespace inlanecruising_plugin
 
         return trajectory;
     }
+
+    tk::spline InLaneCruisingPlugin::compute_fit(std::vector<lanelet::BasicPoint2d> basic_points){
+        int n = int(basic_points.size())/3;
+
+        tk::spline spl;
+
+        std::vector<double> points_x;
+        points_x.push_back(basic_points[n].x());
+        points_x.push_back(basic_points[2*n].x());
+        points_x.push_back(basic_points[3*n].x());
```

Comment:
This will seg fault for even numbers
If size = 12 for a vector with indexes 0-11
12 / 3 = 4
n = 4, n*2 = 8, n*3 = 12

Maybe wrap this in a min call
[std::min(basic_points.size() - 1, n*3)]